core/utils.h

```diff
@@ -1427,7 +1427,8 @@ extern mutex_t do_threshold_mutex;
          */                                                                            \
         ASSERT(                                                                        \
             (try_pointer) == &global_try_except || (try_pointer) == NULL ||            \
-            (try_pointer) == &get_thread_private_dcontext()->try_except ||             \
+            (get_thread_private_dcontext() != NULL &&                                  \
+             (try_pointer) == &get_thread_private_dcontext()->try_except) ||           \
```

Comment:
When the null sanitizer is enabled, this raises an exception.

Reply:
The crash without this fix:
```
Thread 2 "<redacted>" received signal SIGILL, Illegal instruction.
[Switching to Thread 0x7ffff793d700 (LWP 1006714)]
0x0000555572607565 in copy_frame_to_stack (dcontext=0x7ffdaedd5840, info=0x7ffdaee6d148, sig=12, frame=0x7ffdaee88378, sp=0x7ffff793b168 "P\207\347\367\377\177", from_pending=0 '\000') at third_party/dynamorio/trunk/core/unix/signal.c:3558
3558	        TRY_EXCEPT(

```
Disassembly for this part, notice the null check for the result of `get_thread_private_dcontext`:
```
   0x55557260754f <copy_frame_to_stack+1423>:	callq  0x5555725811c0 <get_thread_private_dcontext>
   0x555572607554 <copy_frame_to_stack+1428>:	mov    %rax,-0x188(%rbp)
   0x55557260755b <copy_frame_to_stack+1435>:	cmp    $0x0,%rax
   0x55557260755f <copy_frame_to_stack+1439>:	jne    0x55557260756a <copy_frame_to_stack+1450>
=> 0x555572607565 <copy_frame_to_stack+1445>:	ud1    0x16(%eax),%eax
   0x55557260756a <copy_frame_to_stack+1450>:	mov    -0x190(%rbp),%rax

```

Reply:
From the disassembly above, I see that you are correct about the result of `get_thread_private_dcontext()` not getting dereferenced, but the null check is still applied to it it seems.

Reply:
Added a comment to clarify this.

Reply:
Looks like a bug in the null sanitizer to me.

Reply:
Probably. We'll still need to add this fix though. The comment should help readers.

Reply:
If this were release-build critical-path code I would push to disable the broken sanitizer rather than add unnecessary overhead.  Since it's debug this seem fine.